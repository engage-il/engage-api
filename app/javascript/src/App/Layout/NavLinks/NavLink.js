// @flow
import * as React from 'react'
import styled, { cx, css } from 'react-emotion'
import { withRouter } from 'react-router-dom'
import type { ContextRouter } from 'react-router-dom'
import { Link } from '@/components'
import type { LinkProps } from '@/components'
import { mixins, colors } from '@/styles'
import { removeRouterProps } from '@/functions'

type Props
  = LinkProps
  & ContextRouter

let NavLink = function NavLink ({ to: url, className, location, ...otherProps }: Props) {
  return (
    <Container
      aria-current={location.pathname === url ? 'page' : false}
    >
      <Link
        {...removeRouterProps(otherProps)}
        to={url}
        className={cx(linkClass, className)}
      />
    </Container>
  )
}

const Container = styled.div`
  display: flex;
  align-items: center;
  position: relative;

  ${mixins.mobile`
    padding-bottom: 10px;
  `}

  &[aria-current='page'] {
    &:after {
      content: '';
      position: absolute;
      width: 100%;
      height: 4px;
      bottom: -2px;
      background-color: ${colors.primary} ;
    }

    > * {
      color: ${colors.primary};
    }
  }
`

const linkClass = css`
  ${css(mixins.fonts.light)};

  font-size: 24px;
  text-transform: uppercase;
  text-decoration: none;
  color: ${colors.black};

  &:hover {
    color: ${colors.primary};
  }
`

NavLink = withRouter(NavLink)

export { NavLink }
