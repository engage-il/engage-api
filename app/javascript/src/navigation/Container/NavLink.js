// @flow
import * as React from 'react'
import { cx, css } from 'react-emotion'
import { withRouter } from 'react-router-dom'
import type { ContextRouter } from 'react-router-dom'
import { Link } from 'shared/components'
import type { LinkProps } from 'shared/components'
import { mixins, colors } from 'shared/styles'
import { withoutRouter } from 'shared/functions'

type Props
  = LinkProps
  & ContextRouter

let NavLink = function NavLink ({ to: url, className, location, ...otherProps }: Props) {
  return (
    <Link
      {...withoutRouter(otherProps)}
      to={url}
      className={cx(linkClass, className)}
      aria-current={location.pathname === url ? 'page' : false}
    />
  )
}

const linkClass = css`
  ${css(mixins.fonts.light)};

  font-size: 24px;
  text-transform: uppercase;
  text-decoration: none;
  color: ${colors.black};

  &:hover {
    color: ${colors.primary};
  }

  &[aria-current='page'] {
    color: ${colors.primary};
  }
`

NavLink = withRouter(NavLink)

export { NavLink }
