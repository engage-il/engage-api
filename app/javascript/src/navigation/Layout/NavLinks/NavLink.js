// @flow
import * as React from 'react'
import styled, { cx, css } from 'react-emotion'
import { withRouter } from 'react-router-dom'
import type { ContextRouter } from 'react-router-dom'
import { Link } from 'shared/components'
import type { LinkProps } from 'shared/components'
import { mixins, colors } from 'shared/styles'
import { withoutRouter } from 'shared/functions'

type Props
  = LinkProps
  & ContextRouter

let NavLink = function NavLink ({ to: url, className, location, ...otherProps }: Props) {
  return (
    <Container
      aria-current={location.pathname === url ? 'page' : false}
    >
      <Link
        {...withoutRouter(otherProps)}
        to={url}
        className={cx(linkClass, className)}
      />
    </Container>
  )
}

const Container = styled.div`
  display: flex;
  align-items: center;
  position: relative;

  ${mixins.mobile`
    padding-bottom: 10px;
  `}

  &[aria-current='page'] {
    &:after {
      content: '';
      position: absolute;
      width: 100%;
      height: 3px;
      bottom: 0;
      background-color: ${colors.primary} ;
    }

    > * {
      color: ${colors.primary};
    }
  }
`

const linkClass = css`
  ${css(mixins.fonts.light)};

  font-size: 24px;
  text-transform: uppercase;
  text-decoration: none;
  color: ${colors.black};

  &:hover {
    color: ${colors.primary};
  }
`

NavLink = withRouter(NavLink)

export { NavLink }
