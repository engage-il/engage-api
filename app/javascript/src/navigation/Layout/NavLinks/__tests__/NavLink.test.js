/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { defaultsDeep } from 'lodash'
import { NavLink } from '../NavLink'

// subject
let subject

const defaults = {
  to: 'test-url-1',
  className: 'test-class',
  location: {
    pathname: 'test-url-2'
  }
}

function loadSubject (props = {}) {
  subject = shallow(<NavLink {...defaultsDeep(props, defaults)} />)
}

// spec
afterEach(() => {
  subject = null
})

describe('#render', () => {
  it('shows the link as inactive', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })

  it('shows the link as active when the page matches', () => {
    loadSubject({
      location: {
        pathname: 'test-url-1'
      }
    })

    expect(subject).toMatchSnapshot()
  })
})
