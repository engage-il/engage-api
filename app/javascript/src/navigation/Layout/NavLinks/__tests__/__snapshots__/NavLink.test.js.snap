// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render shows the link as active when the page matches 1`] = `
<Styled(div)
  aria-current="page"
>
  <Link
    className="css-1t0ocqs test-class"
    to="test-url-1"
  />
</Styled(div)>
`;

exports[`#render shows the link as inactive 1`] = `
<Styled(div)
  aria-current={false}
>
  <Link
    className="css-1t0ocqs test-class"
    to="test-url-1"
  />
</Styled(div)>
`;
